share/functions/fish_svn_prompt.fish

```diff
@@ -70,15 +70,15 @@ set -g __fish_svn_prompt_char_token_broken_color --bold magenta
 function __fish_svn_prompt_parse_status --argument flag_status_string --description "helper function that does pretty formatting on svn status"
     # SVN status symbols
     # Do not change these! These are the expected characters that are output from `svn status`, they are taken from `svn help status`
-    set -l __fish_svn_prompt_chars A C D I M R X \? ! ~ L + S K O T B
+    set -l __fish_svn_prompt_chars A C D I M R X \\\? ! \~ L + S K O T B
```

Comment:
The `?` should be *quoted*, not escaped - currently the `?` can be disabled via the "qmark-noglob" feature flag (which will be turned on and eventually removed in future), so how many backslashes you need depends on the setting.

Quoting it should always work.

Reply:
I'm not sure if quoting even works - `string match` will change behavior depending on `status test-feature qmark-noglob`.

```fish
~> fish_features= fish -c "string match '?' x"
x
~> fish_features=qmark-noglob fish -c "string match '?' x"
~ [1]> 
```

`string match --regex` or `contains` should work

Reply:
Okay, I wasn't aware of the qmark-noglob feature. And no, just quoting it isn't enough.
So maybe this isn't the best way to go ...
Can `contains` be used on a string? I assumed it was meant for lists.

Reply:
> Can contains be used on a string? I assumed it was meant for lists.

Yep, I believe that

`        if test (count (string match $__fish_svn_prompt_chars[$flag_index] -- $flag_status_string)) -eq 1
`

can be just

`
        if contains $__fish_svn_prompt_chars[$flag_index] -- $flag_status_string
`

assuming that `$flag_status_string` is a list, otherwise even `test x = y` could work.

Reply:
> can be just
> 
> `if contains $__fish_svn_prompt_chars[$flag_index] -- $flag_status_string`
> 
> assuming that `$flag_status_string` is a list,

It is not :-)

> otherwise even `test x = y` could work.

Won't work either, because, like I described below, we need to test if x is a substring of y (or more specifically, if the character x is contained in the string y).

Reply:
`$flag_status_string` should probably just be a list. These are single characters, so I don't see why we join them.